dotnet/src/SemanticKernel.Abstractions/Filters/Function/FunctionInvokedContext.cs

```diff
@@ -23,11 +23,29 @@ public FunctionInvokedContext(KernelArguments arguments, FunctionResult result)
         this.ResultValue = result.Value;
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FunctionInvokedContext"/> class.
+    /// </summary>
+    /// <param name="arguments">The arguments associated with the operation.</param>
+    /// <param name="result">The result of the function's invocation.</param>
+    /// <param name="exception">The exception that occurred during function invocation.</param>
+    public FunctionInvokedContext(KernelArguments arguments, FunctionResult result, Exception exception)
+        : this(arguments, result)
+    {
+        this.Exception = exception;
+    }
+
     /// <summary>
     /// Gets the result of the function's invocation.
     /// </summary>
     public FunctionResult Result { get; }
 
+    /// <summary>
+    /// Gets the exception that occurred during function invocation.
+    /// If it's <see langword="null" />, it means that function invocation was completed successfully.
+    /// </summary>
+    public Exception? Exception { get; private set; }
```

Comment:
I thought about that, but I'm wondering if this will be useful. If we want to change the exception, it should be possible to `throw` it directly from filter, instead of assigning it to property. I will double check if it's supported.

Reply:
> we will allow function filters to change the exceptions thrown.

I added test and example which show that it's possible without setter on `Exception` property. Basically, if you need to change the exception, you can simple `throw` another exception directly in filter.